### PR TITLE
Fix accidentally inverted condition for room ordering

### PR DESCRIPTION
**Pull request by justjanne**
Type: Defect
Fixes: https://github.com/vector-im/element-web/issues/24527

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix accidentally inverted condition for room ordering ([\#10178](https://github.com/matrix-org/matrix-react-sdk/pull/10178)). Fixes vector-im/element-web#24527. Contributed by @justjanne.

### CODE REVIEW

**Review by t3chguy — APPROVED** [review]

**Review by tulir — APPROVED** [review]
My room list is no longer completely messed up 🎉

File: src/stores/room-list/algorithms/list-ordering/ImportanceAlgorithm.ts [hunk 1]
```diff
@@ -104,7 +104,6 @@ export class ImportanceAlgorithm extends OrderingAlgorithm {
             // Every other sorting type affects the categories, not the whole tag.
             const categorized = this.categorizeRooms(rooms);
             for (const category of Object.keys(categorized)) {
-                if (!isNaN(Number(category))) continue;
```
Comment (MadLittleMods) [inline src/stores/room-list/algorithms/list-ordering/ImportanceAlgorithm.ts:107]: Can we add a test for this? Just a general rooms should be sorted as expected test.

Comment (MadLittleMods) [inline src/stores/room-list/algorithms/list-ordering/ImportanceAlgorithm.ts:107]: Why was this line added before? Did it ever make sense?

Why can we remove it now?

Let's add this context to the PR description
